packages/Search/examples/bvh_driver/bvh_driver.cpp

```diff
@@ -95,110 +47,99 @@ int main_( Teuchos::CommandLineProcessor &clp, int argc, char *argv[] )
-        {
-            k_host[i] = distribution_k( generator );
-        }
-        Kokkos::deep_copy( k, k_host );
-
-        Kokkos::View<details::Nearest<DataTransferKit::Point> *, DeviceType>
-            nearest_queries( "nearest_queries", n_points );
+        Kokkos::View<
+            DataTransferKit::Details::Nearest<DataTransferKit::Point> *,
+            DeviceType>
+            queries( "queries", n_queries );
         Kokkos::parallel_for(
-            DTK_MARK_REGION( "register_nearest_queries" ),
-            Kokkos::RangePolicy<ExecutionSpace>( 0, n_points ),
+            Kokkos::RangePolicy<ExecutionSpace>( 0, n_queries ),
             KOKKOS_LAMBDA( int i ) {
-                nearest_queries( i ) = details::nearest<DataTransferKit::Point>(
-                    {{point_coords( i, 0 ), point_coords( i, 1 ),
-                      point_coords( i, 2 )}},
-                    k( i ) );
+                queries( i ) =
+                    DataTransferKit::Details::nearest<DataTransferKit::Point>(
+                        random_points( i ), 10 );
             } );
         Kokkos::fence();
 
-        // do the search
-        Kokkos::View<int *, DeviceType> offset_nearest( "offset_nearest" );
-        Kokkos::View<int *, DeviceType> indices_nearest( "indices_nearest" );
-        bvh.query( nearest_queries, indices_nearest, offset_nearest );
+        Kokkos::View<int *, DeviceType> offset( "offset" );
+        Kokkos::View<int *, DeviceType> indices( "indices" );
+        start = std::chrono::high_resolution_clock::now();
+        bvh.query( queries, indices, offset );
+        end = std::chrono::high_resolution_clock::now();
+        elapsed_seconds = end - start;
+        os << "knn " << elapsed_seconds.count() << "\n";
     }
-    else if ( mode == "radius" )
-    {
-        Kokkos::View<double *, ExecutionSpace> radii( "radii", n_points );
-        auto radii_host = Kokkos::create_mirror_view( radii );
 
-        // use random radius for the search
-        // set the limit of approximately 100 points by
-        // solving n_points*pi*r^2/(Lx^2 + Ly^2 + Lz^2) <= 100
-        const int approx_points = 100;
-        double max_radius = sqrt(
-            approx_points * ( Lx * Lx + Ly * Ly + Lz * Lz ) / ( n * M_PI ) );
-        std::uniform_real_distribution<double> distribution_radius(
-            0.0, max_radius );
-        for ( int i = 0; i < n_points; ++i )
-        {
-            radii_host[i] = distribution_radius( generator );
-        }
-        Kokkos::deep_copy( radii, radii_host );
-
-        Kokkos::View<details::Within *, DeviceType> within_queries(
-            "within_queries", n_points );
+    {
+        Kokkos::View<DataTransferKit::Details::Within *, DeviceType> queries(
+            "queries", n_queries );
+        // radius chosen such that there will be approximately 10 results per
+        // query
+        double const pi = 3.14159265359;
+        double const r = std::cbrt( 10. * 3. / ( 4. * pi ) );
```

Comment:
Given that the input point set is random I think this is about as good as you can do to get an equivalent number of numbers from both kNN and radius search.